src/gobstuf/mks_utils.py

```diff
@@ -47,37 +127,106 @@ def _dd(cls, mks_datum):
         return mks_datum[6:8] if cls._is_mks_datum(mks_datum) else None
 
     @classmethod
-    def as_datum(cls, mks_datum):
+    def as_datum_broken_down(cls, mks_datum, ind_onvolledige_datum=None):
         if cls._is_mks_datum(mks_datum):
+            return {
+                'datum': cls.as_datum(mks_datum, ind_onvolledige_datum),
+                'jaar': cls.as_jaar(mks_datum, ind_onvolledige_datum),
+                'maand': cls.as_maand(mks_datum, ind_onvolledige_datum),
+                'dag': cls.as_dag(mks_datum, ind_onvolledige_datum)
+            }
+
+    @classmethod
+    def as_datum(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_datum_volledig():
             return f"{cls._yyyy(mks_datum)}-{cls._mm(mks_datum)}-{cls._dd(mks_datum)}"
 
     @classmethod
-    def as_jaar(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_jaar(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_jaar_bekend():
             return int(cls._yyyy(mks_datum))
 
     @classmethod
-    def as_maand(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_maand(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_maand_bekend():
             return int(cls._mm(mks_datum))
 
     @classmethod
-    def as_dag(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_dag(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_dag_bekend():
             return int(cls._dd(mks_datum))
 
     @classmethod
-    def as_leeftijd(cls, mks_geboortedatum):
+    def _get_age(cls, now, birthday):
+        """
+        Age calculation that takes leap years into account
+
+        :param now:
+        :param birthday:
+        :return:
+        """
+        try:
+            day = birthday.replace(year=now.year)
+        except ValueError:
```

Comment:
Bit of a wide exception.

Using calendar.isleapyear() with check for month and day is more explicit and cleaner I think.

Reply:
Nice remark! Will check on it

Reply:
Much easier to read now